lib/github-pages-health-check/domain.rb

```diff
@@ -381,7 +381,11 @@ def caa_error
       private
 
       def caa
-        @caa ||= GitHubPages::HealthCheck::CAA.new(host, :nameservers => nameservers)
+        @caa ||= if cname?
```

Comment:
As per https://www.rfc-editor.org/errata/eid5065, we'll need to still check the CAA records for the original hostname and respect any CAA records here first, before then checking the target of the CNAME if no CAA records are found for the original hostname.